test/rust_src/src/data-tests.el

```diff
@@ -36,5 +36,54 @@
   (should-error (aref 100 1)
                 :type 'wrong-type-argument))
 
+(ert-deftest data-test--aset-base ()
+  "Verify (aset) base cases"
+  (should-error (aset "abc" -1 ?d) :type 'args-out-of-range)
+  (should-error (aset 100 1 1)
+                :type 'wrong-type-argument)
+  (let ((object "abcdef"))
+    (should (eq (aset object 3 ?w)
+                ?w))
+    (should (eq (aref object 3)
+                ?w)))
+  (should-error (aset "abc" 3 ?w)
+                :type 'args-out-of-range)
+  (let ((bv (bool-vector t nil t nil)))
+    (should-error (aset bv 4 t)
+                  :type 'args-out-of-range)
+    (should (eq (aset bv 1 t)
+                t))
+    (should (eq (aref bv 1)
+                t)))
+  (let ((tbl (make-char-table 'foo 0)))
+    (should (eq (aset tbl 0 1)
+                1))
+    (should (eq (aref tbl 0)
+                1)))
+  (let* ((x 1)
+         (y 2)
+         (z 3)
+         (a 10)
+         (v [x y z]))
```

Comment:
Vectors literals do not evaluate their arguments, so this is equivalent to `(vector 'x 'y 'z)`. You're not using x, y or z as far as I can see. Perhaps just write `[1 2 3]`?

Reply:
I used variables to avoid magic numbers. I can expect `y` instead of `2`. But I am not good at idiomatic elisp. Whichever style makes the most sense I can use here.